Add note on use of `<use>` (#1050)
lucide-icons/lucide: docs/icon-design-guide.md

Before an icon is added to the library, we like to have readable and optimized svg code.

> Don't use [`<use>`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use). While it may seem like a good way to optimize file size sometimes,there's no way to ensure that the referenced element IDs are unique when SVGs are embedded in HTML documents.

Reviewer: Great idea, thanks, I've rephrased some parts.

I think in the long run, we're going to have to explicitly state (and possibly even enforce) which attributes on which elements are allowed at all.

```suggestion
Never use [`<use>`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use). While it may sometimes seem like a good way to optimize file size, there's no way to ensure that the referenced element IDs will be unique once the SVGs are embedded in HTML documents.
```
